logger.go

```diff
@@ -84,13 +109,21 @@ func (l *MicroLogger) LogCtx(ctx context.Context, keyVals ...interface{}) {
 	l.log(keyValsWithMeta(ctx, keyVals))
 }
 
-func (l *MicroLogger) With(keyVals ...interface{}) Logger {
-	keyVals = processStack(keyVals)
+func (l *MicroLogger) deepCopy() *MicroLogger {
 	return &MicroLogger{
-		logger: kitlog.With(l.logger, keyVals...),
+		info:      l.info,
+		logger:    l.logger,
```

Comment:
This isn't technically a deep copy, but `go-kit/log` appears to be immutable so I think it's close enough.